src/Biogeochemistry.jl

```diff
@@ -34,25 +34,32 @@ update_biogeochemical_state!(bgc, model) = nothing
 Abstract type for biogeochemical models. To define a biogeochemcial relaionship
 the following functions must have methods defined where `BiogeochemicalModel`
 is a subtype of `AbstractBioeochemistry`:
- - `(bgc::BiogeochemicalModel)(i, j, k, grid, ::Val{:TRACER_NAME}, clock, fields)` which 
-    returns the biogeochemical reaction for for each tracer
- - `required_biogeochemical_tracers(::BiogeochemicalModel)` which returns a tuple of
-    required tracer names
- - `required_biogeochemical_auxiliary_fields(::BiogeochemicalModel)` which returns 
-    a tuple of required auxiliary fields
- - `biogeochemical_auxiliary_fields(bgc::BiogeochemicalModel)` which returns a `NamedTuple`
-    of the models auxiliary fields (e.g. `(PAR = bgc.light_attenuation.PAR_field, )`)
- - `biogeochemical_drift_velocity(bgc::BiogeochemicalModel, ::Val{:TRACER_NAME})` which 
-    returns a velocity fields (i.e. a `NamedTuple` of fields with keys `u`, `v` & `w`)
-    for each tracer
- - `biogeochemical_advection_scheme(bgc::BiogeochemicalModel, ::Val{:TRACER_NAME})` which
-    returns an advection scheme for each tracer.
- - `update_biogeochemical_state!(bgc::BiogeochemicalModel, model)` (optional) to update the
-    model state
+
+    - `(bgc::BiogeochemicalModel)(i, j, k, grid, ::Val{:tracer_name}, clock, fields)` which 
+       returns the biogeochemical reaction for for each tracer.
+
+    - `required_biogeochemical_tracers(::BiogeochemicalModel)` which returns a tuple of
+       required `tracer_names`.
+
+    - `required_biogeochemical_auxiliary_fields(::BiogeochemicalModel)` which returns 
+       a tuple of required auxiliary fields.
+
+    - `biogeochemical_auxiliary_fields(bgc::BiogeochemicalModel)` which returns a `NamedTuple`
+       of the models auxiliary fields.
+
+    - `biogeochemical_drift_velocity(bgc::BiogeochemicalModel, ::Val{:tracer_name})` which 
+       returns a velocity fields (i.e. a `NamedTuple` of fields with keys `u`, `v` & `w`)
+       for each tracer.
+
+    - `biogeochemical_advection_scheme(bgc::BiogeochemicalModel, ::Val{:tracer_name})` which
+       returns an advection scheme for each tracer.
+
+    - `update_biogeochemical_state!(bgc::BiogeochemicalModel, model)` (optional) to update the
+        model state.
 """
 abstract type AbstractBiogeochemistry end
 
-@inline biogeochemistry_rhs(i, j, k, grid, ::Nothing, val_tracer_name::Val{tracer_name}, clock, fields) where tracer_name = zero(grid)
```

Comment:
I think I had to specify it like this to stop the tests failing like they did here